Tasks/AzureFunctionAppV1/deploymentProvider/ConsumptionWebAppDeploymentProvider.ts

```diff
@@ -19,8 +19,14 @@ export class ConsumptionWebAppDeploymentProvider extends AzureRmWebAppDeployment
         let appSettings = await this.appService.getApplicationSettings();
         var storageData = {};
         if(appSettings && appSettings.properties && appSettings.properties.AzureWebJobsStorage) {
-            let str = appSettings.properties.AzureWebJobsStorage;
-            let dictionary = getKeyValuePairs(str);
+            let webStorageSetting = appSettings.properties.AzureWebJobsStorage;
+            if (!webStorageSetting) {
+                throw new Error(tl.loc('FailedToGetStorageAccountDetails'));
+            }
+            let dictionary = getKeyValuePairs(webStorageSetting);
+            if (!dictionary["AccountName"] || !dictionary["AccountKey"]) {
```

Comment:
cleanup. Have a if block which fetches data from app setting. followed by if block that throws error if storage account details are not fetched.

Reply:
It is done.